src/snowflake/snowpark/_internal/analyzer/datatype_mapper.py

```diff
@@ -226,6 +233,14 @@ def to_sql(
     if isinstance(datatype, VectorType):
         return f"{value} :: VECTOR({datatype.element_type},{datatype.dimension})"
 
+    if isinstance(datatype, FileType):
+        # TODO: SNOW-1950688: Remove parsing workaround once the server is ready for accepting full stage URI
+        parts = validate_stage_location(str(value)).split("/", maxsplit=1)
+        if len(parts) != 2:
```

Comment:
Does it have to be exactly 2? I think you need to find the first `/` if the length is > 2.

Reply:
Forget it. There is maxsplit.